### PR TITLE
gh-92417: `subprocess` docs: remove note on compatibility with Python <3.5

### PR DESCRIPTION
**Pull request by AlexWaygood**
#92417

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
🧹

File: Doc/library/subprocess.rst [hunk 1]
```diff
@@ -33,9 +33,6 @@ The recommended approach to invoking subprocesses is to use the :func:`run`
 function for all use cases it can handle. For more advanced use cases, the
 underlying :class:`Popen` interface can be used directly.
 
-The :func:`run` function was added in Python 3.5; if you need to retain
```
Comment (rhettinger) [inline Doc/library/subprocess.rst:36]: We normally keep the notes. on when a function was added.

There are many packages that need to maintain cross-version support even if we don't support the old versions.  This information is critical for them.

Comment (AlexWaygood) [reply Doc/library/subprocess.rst:36]: The `.. versionadded:: 3.5` note for `subprocess.run()`, on line 103 of this file, is not touched by this PR (and I would not want to remove it). This PR proposes to remove an additional note, which seems to me to be overly prominent now that 3.4 has been end-of-life for a long time.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by ambv** [comment]
Backporting to 3.10 for consistency with comment in https://github.com/python/cpython/pull/92544#issuecomment-1131867041.

**Pull request merged**